src/GafferImage/Constant.cpp

```diff
@@ -141,17 +148,26 @@ IECore::ConstCompoundObjectPtr Constant::computeMetadata( const Gaffer::Context
 void Constant::hashChannelNames( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
 {
 	ImageNode::hashChannelNames( output, context, h );
+	layerPlug()->hash( h );
 }
 
 IECore::ConstStringVectorDataPtr Constant::computeChannelNames( const Gaffer::Context *context, const ImagePlug *parent ) const
 {
-	IECore::StringVectorDataPtr channelStrVectorData( new IECore::StringVectorData() );
-	std::vector<std::string> &channelStrVector( channelStrVectorData->writable() );
-	channelStrVector.push_back("R");
-	channelStrVector.push_back("G");
-	channelStrVector.push_back("B");
-	channelStrVector.push_back("A");
-	return channelStrVectorData;
+	std::string channelNamePrefix = layerPlug()->getValue();
```

Comment:
I don't feel strongly about this at all, and I wrote that before Christmas so don't remember my thinking. But if I were to pull a reason out of the air now I'd say that it's because by the time we've appended the ".", it's not quite a layer name any more...